### DISCUSSION TITLE
Wait for message window to be "open" before sending message.

### DISCUSSION THREAD

**Discussion by esbe1175** [Chatterino/chatterino2#4032] [Ideas] [answered]
as mentioned in [#1213](https://github.com/Chatterino/chatterino2/issues/1213) and [#1512](https://github.com/Chatterino/chatterino2/issues/1512), messages do not send unless the chat is "Moving", because of how twitch optimizes/limits the chat server-side.

I propose some sort of toggle that makes it so chatterino waits with sending your message until there is a reasonable likelihood that it will go through. I feel like if I can sit here as a human being, waiting for the chat to move before sending it, there's no reason we can't have an automated software feature to do that part for us.

Hope this is possible. Recently been watching some streams where it was virtually impossible to even get a message through.

**Comment by Felanbird** [comment, marked as answer]
> there's no reason we can't have an automated software feature to do that part for us.

Yes there is, this would be classified as scripting to guarantee your message goes through, Chatterino is not a botting/scripting application.
It would be nice if Twitch could handle the amount of messages they get, but that's not on us to fix.

**Reply by esbe1175** [reply]
I see, i guess there is a design philosphy behind chatterino i didn't really think of in that case. Twitch has been triggerhappy with site changes recently, so who knows 🤷‍♂️
